src/main/java/emissary/core/channels/BufferedChannelFactory.java

```diff
@@ -0,0 +1,132 @@
+package emissary.core.channels;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.Validate;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.SeekableByteChannel;
+
+/**
+ * Creates a SeekableByteChannel cache of a defined size analogous to BufferedInputStream.
+ */
+public final class BufferedChannelFactory {
+    private BufferedChannelFactory() {}
+
+    /**
+     * Creates a SeekableByteChannelFactory that caches the bytes of the passed in SeekableByteChannelFactory.
+     * 
+     * @param seekableByteChannelFactory to be cached.
+     * @param blocksize of the block of bytes.
+     * @return the caching SeekableByteChannelFactory.
+     */
+    public static SeekableByteChannelFactory create(final SeekableByteChannelFactory seekableByteChannelFactory,
+            final int blocksize) {
```

Comment:
Suggest calling this `bufferSize` or `cacheSize`

Reply:
Changed to bufferSize.